pkgs/applications/virtualization/xen/generic.nix

```diff
@@ -1,265 +1,376 @@
+            chmod -R +w tools/${name}
+          ''
+        )}
+      '';
+
+    patches = config.patches or [ ];
+
+    postPatch =
+      # The following patch forces Xen to install xen.efi on $out/boot
+      # instead of $out/boot/efi/efi/nixos, as the latter directory
+      # would otherwise need to be created manually.
+      ''
+        substituteInPlace xen/Makefile \
+          --replace-fail "\$(D)\$(EFI_MOUNTPOINT)/efi/\$(EFI_VENDOR)/\$(T)-\$(XEN_FULLVERSION).efi" \
+                    "\$(D)\$(BOOT_DIR)/\$(T)-\$(XEN_FULLVERSION).efi"
+      ''
+
+      # The following patch fixes the call to /bin/mkdir on the
+      # launch_xenstore.sh helper script.
+      + ''
+        substituteInPlace tools/hotplug/Linux/launch-xenstore.in \
+          --replace-fail "/bin/mkdir" "${coreutils}/bin/mkdir"
+      ''
+
+      # The following expression fixes the paths called by Xen's systemd
+      # units, so we can use them in the NixOS module.
+      + ''
+        substituteInPlace \
+          tools/hotplug/Linux/systemd/{xen-init-dom0,xen-qemu-dom0-disk-backend,xenconsoled,xendomains,xenstored}.service.in \
+          --replace-fail /bin/grep ${gnugrep}/bin/grep
+        substituteInPlace \
+         tools/hotplug/Linux/systemd/{xen-qemu-dom0-disk-backend,xenconsoled}.service.in \
+          --replace-fail "/bin/mkdir" "${coreutils}/bin/mkdir"
+      ''
+
+      # The following expression applies the patches defined on each
+      # prefetchedSources attribute.
+      + ''
+        ${withTools "patches" (
+          name: pkg: ''
+            ${lib.strings.concatMapStringsSep "\n" (patch: ''
+              echo "Patching with ${patch}"
+              patch -p1 < ${patch}
+            '') pkg.patches}
+          ''
+        )}
+
+             ${withTools "postPatch" (name: pkg: pkg.postPatch)}
+
+             ${config.postPatch or ""}
+      '';
+
+    makeFlags = [
+      "XSM_ENABLE=y"
+      "FLASK_ENABLE=y"
+
+      "EFI_VENDOR=nixos"
+      "LD=${efiBinutils}/bin/ld"
+
+      "PREFIX=$(out)"
+      "CONFIG_DIR=/etc"
+      "XEN_EXTFILES_URL=\\$(XEN_ROOT)/xen_ext_files"
+      "XEN_SCRIPT_DIR=$(CONFIG_DIR)/xen/scripts"
+      "BASH_COMPLETION_DIR=$(PREFIX)/share/bash-completion/completions"
+    ] ++ (config.makeFlags or [ ]);
+
+    preBuild = ''
+      ${config.preBuild or ""}
+    '';
+
+    buildFlags = [
+      "xen"
+      "tools"
     ];
-  } // (config.meta or {});
-} // removeAttrs config [ "xenfiles" "buildInputs" "patches" "postPatch" "meta" ])
+
+    postBuild = ''
+      make -C docs man-pages
+
+      ${withTools "buildPhase" (name: pkg: pkg.buildPhase)}
+    '';
+
+    installPhase =
+      # Create $out directories and copy build output.
+      ''
+        mkdir -p $out $out/share $out/share/man
+        cp -prvd dist/install/nix/store/*/* $out/
+        cp -prvd dist/install/boot $out/boot
+        cp -prvd dist/install/etc $out
+        cp -dR docs/man1 docs/man5 $out/share/man/
```

Comment:
Use full names to make the derivation easier to read instead of shorthands?

Reply:
Those lines are kept from the previous commit (I guess the diff says I added them because of `nixfmt`.)

Could you elaborate on what exactly is hard to read here? Seems self-explanatory to me that line 306 copies the binaries and libraries, line 307 copies the kernel and EFI file, line 308 copies the Xen configuration defaults, and like 309 copies the manual.

Reply:
I am aware that this is from the previous commit and i wouldn't say hard to read rather that e.g. `cp --parents --recursive` is more understandable to the common contributor than `cp -pr` imho

Reply:
Ahh, the flags; I thought you were talking about the /nix/store/*/* wildcard.

I agree completely with this change. Added to the queue.